server/graphql/v2/query/ExpensesQuery.ts

```diff
@@ -77,6 +81,25 @@ const ExpensesQuery = {
       where['CollectiveId'] = collective.id;
     }
 
+    // Add search filter
+    if (args.searchTerm) {
+      where[Op.or] = [
+        { description: { [Op.substring]: args.searchTerm }},
+        { tags: { [Op.overlap]: [args.searchTerm] }},
+        { '$fromCollective.name$': { [Op.substring]: args.searchTerm }},
+        { '$items.description$': { [Op.substring]: args.searchTerm }}
+      ];
```

Comment:
`Op.substring` looks very clean, but we need to be case-insensitive (ie. searching `food` should return titles like `Food for the team`). I can suggest:

```suggestion
      const sanitizedTerm = args.searchTerm.replace(/(_|%|\\)/g, '\\$1');
      const ilikeQuery = `${sanitizedTerm}%`;
      where[Op.or] = [
        { description: { [Op.iLike]: ilikeQuery } },
        { tags: { [Op.overlap]: [args.searchTerm.toLowerCase()] } },
        { '$fromCollective.slug$': { [Op.iLike]: ilikeQuery } },
        { '$fromCollective.name$': { [Op.iLike]: ilikeQuery } },
        { '$items.description$': { [Op.iLike]: ilikeQuery } },
      ];
```

Reply:
@Betree : I think line 87 should be, `const ilikeQuery = `%${sanitizedTerm}%`;` since otherwise we are missing some search results. Some strings in the middle of sentences will be omitted. This maybe the reason that you see less than 5 results all the time. I've pushed these changes; let me know if you agree. :+1:

Reply:
Good catch